tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/AUC.java

```diff
@@ -0,0 +1,1021 @@
+        throw new IllegalArgumentException(
+            String.format(
+                "labels must have rank=2 when multiLabel is true. Found rank %d.",
+                shape.numDimensions()));
+      this.numLabels = (int) shape.size(1);
+      variableShape = Shape.of(this.numThresholds, this.numLabels);
+    } else {
+      variableShape = Shape.of(this.numThresholds);
+    }
+
+    Zeros<T> zeros = new Zeros<>(getTF());
+    Operand<T> zero = zeros.call(getTF().constant(variableShape), type);
+    if (truePositives == null) {
+      truePositives = getTF().withName(getTruePositivesName()).variable(zero);
+      initializers.put(ConfusionMatrixEnum.TRUE_POSITIVES, getTF().assign(truePositives, zero));
+    }
+
+    if (falsePositives == null) {
+      falsePositives = getTF().withName(getFalsePositivesName()).variable(zero);
+      initializers.put(ConfusionMatrixEnum.FALSE_POSITIVES, getTF().assign(falsePositives, zero));
+    }
+
+    if (trueNegatives == null) {
+      trueNegatives = getTF().withName(getTrueNegativesName()).variable(zero);
+      initializers.put(ConfusionMatrixEnum.TRUE_NEGATIVES, getTF().assign(trueNegatives, zero));
+    }
+
+    if (falseNegatives == null) {
+      falseNegatives = getTF().withName(getFalseNegativesName()).variable(zero);
+      initializers.put(ConfusionMatrixEnum.FALSE_NEGATIVES, getTF().assign(falseNegatives, zero));
+    }
+
+    this.initialized = true;
+    return initializers;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  @SuppressWarnings("unchecked")
+  public List<Op> updateStateList(
+      Operand<? extends TNumber> labels,
+      Operand<? extends TNumber> predictions,
+      Operand<? extends TNumber> sampleWeights) {
+
+    Operand<T> lLabels = cast(getTF(), labels, type);
+    Operand<T> lPredictions = cast(getTF(), predictions, type);
+    Operand<T> tSampleWeights = sampleWeights != null ? cast(getTF(), sampleWeights, type) : null;
+    List<Op> updateOperations = new ArrayList<>();
+    Map<ConfusionMatrixEnum, Assign<T>> varInitializers = Collections.EMPTY_MAP;
+    if (!this.initialized) {
+      varInitializers = build(lPredictions.shape());
+    }
+    if (this.isMultiLabel() || this.getLabelWeights() != null) {
+      List<SymbolicShape<? extends TNumber>> symbols = new ArrayList<>();
+      symbols.add(new SymbolicShape<>(lLabels, "N", "L"));
+      if (this.isMultiLabel()) {
+        symbols.add(new SymbolicShape<>(this.truePositives, "T", "L"));
+        symbols.add(new SymbolicShape<>(this.falsePositives, "T", "L"));
+        symbols.add(new SymbolicShape<>(this.trueNegatives, "T", "L"));
+        symbols.add(new SymbolicShape<>(this.falseNegatives, "T", "L"));
+      }
+      if (this.getLabelWeights() != null) {
+        symbols.add(new SymbolicShape<>(this.getLabelWeights(), "L", ""));
+      }
+      updateOperations.addAll(
+          MetricsHelper.assertShapes(getTF(), symbols, "Number of labels is not consistent."));
+    }
+    if (this.isMultiLabel()) {
```

Comment:
Ah, I think setting `labelWeights` to `null` is wrong anyway. In the Python code, they don't change the instance variable to `None`. Rather, they create a local that may be `None` just for the purpose of invoking `metrics_utils.update_confusion_matrix_variables`:
~~~Python
    # Only forward label_weights to update_confusion_matrix_variables when
    # multi_label is False. Otherwise the averaging of individual label AUCs is
    # handled in AUC.result
    label_weights = None if self.multi_label else self.label_weights
~~~